src/commands/call.ts

```diff
@@ -139,4 +126,73 @@ export class CallCommand extends ApifyCommand<typeof CallCommand> {
             process.exitCode = CommandExitCodes.RunFailed;
         }
     }
+
+    private async resolveActorId(client: ApifyClient, localConfigName: string | undefined, usernameOrId: string) {
+        const { actorId } = this.args;
+
+        // Full ID
+        if (actorId?.includes('/')) {
+            const actor = await client.actor(actorId).get();
+            if (!actor) throw new Error(`Cannot find Actor with ID '${actorId}' in your account.`);
+
+            return {
+                userFriendlyId: actor.username ? `${actor.username}/${actor.name}` : actor.id,
+                id: actor.id,
+            };
+        }
+
+        // Fetch all actors the user has, and see if any of them match the name
+        if (actorId) {
+            const allActors = await this.fetchAllActors(client);
+
+            const actor = allActors.find((a) => a.name.toLowerCase() === actorId.toLowerCase());
+
+            if (!actor) {
+                throw new Error(`Cannot find Actor with name '${actorId}' in your account.`);
+            }
+
+            return {
+                userFriendlyId: actor.username ? `${actor.username}/${actor.name}` : actor.id,
+                id: actor.id,
+            };
+        }
+
+        if (localConfigName) {
+            const actor = await client.actor(`${usernameOrId}/${localConfigName}`).get();
+
+            if (!actor) {
+                throw new Error(`Cannot find Actor with ID '${usernameOrId}/${localConfigName}' `
+                    + 'in your account. Call "apify push" to push this Actor to Apify platform.');
+            }
+
+            return {
+                userFriendlyId: `${actor.username}/${actor.name}`,
+                id: actor.id,
+            };
+        }
+
+        throw new Error('Please provide an Actor ID or name, or run this command from a directory with a valid Apify actor.');
+    }
+
+    private async fetchAllActors(client: ApifyClient) {
+        let offset = 0;
+
+        const actors: ActorCollectionListItem[] = [];
+
+        do {
+            const { items } = await client.actors().list({ limit: 1000, offset, my: true });
```

Comment:
can't we query the actors by name instead of loading them to memory? or just use the "get single actor" endpoint directly (with the username prefix)? this feels very inefficient

Reply:
Oh shit, doy, good point. Brain didn't brain yesterday 😂